src/main/java/teammates/common/datatransfer/GeneralLogEntry.java

```diff
@@ -0,0 +1,132 @@
+package teammates.common.datatransfer;
+
+import javax.annotation.Nullable;
+
+import org.json.JSONObject;
+
+import com.google.cloud.logging.Payload;
+
+public class GeneralLogEntry {
+    private final String logName;
+    private final String severity;
+    private final String trace;
+    private final SourceLocation sourceLocation;
+    private final Payload<?> payload;
+    private final long timestamp;
+    @Nullable
+    private JSONObject jsonObject;
+
+    public GeneralLogEntry(String logName,
+                           String severity,
+                           String trace,
+                           SourceLocation sourceLocation,
+                           Payload<?> payload,
+                           long timestamp) {
+        this.logName = logName;
+        this.severity = severity;
+        this.trace = trace;
+        this.sourceLocation = sourceLocation;
+        this.payload = payload;
+        this.timestamp = timestamp;
+    }
+
+    public GeneralLogEntry(String logName,
+                            String severity,
+                            String trace,
+                            SourceLocation sourceLocation,
+                            Payload<?> payload,
+                            long timestamp,
+                            JSONObject jsonObject) {
+        this.logName = logName;
+        this.severity = severity;
+        this.trace = trace;
+        this.sourceLocation = sourceLocation;
+        this.payload = payload;
+        this.timestamp = timestamp;
+        this.jsonObject = jsonObject;
+    }
+
+    public String getLogName() {
+        return logName;
+    }
+
+    public String getSeverity() {
+        return severity;
+    }
+
+    public String getTrace() {
+        return trace;
+    }
+
+    public SourceLocation getSourceLocation() {
+        return sourceLocation;
+    }
+
+    public Payload<?> getPayload() {
+        return payload;
+    }
+
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    public JSONObject getJsonObject() {
+        return jsonObject;
+    }
+
+    @Override
+    public String toString() {
+        return "{\n"
+                + "  LogName: " + logName + ",\n"
+                + "  Severity: " + severity + ",\n"
+                + "  Trace: " + trace + ",\n"
+                + "  SourceLocation: " + sourceLocation + ",\n"
+                + "  Payload: " + payload + ",\n"
+                + "  Timestamp: " + timestamp + "\n"
+                + "}";
+    }
+
+    public static class SourceLocation {
+        private final String file;
+        private final Long line;
```

Comment:
It cannot be `null` unless it is intentionally set to `null`.

Reply:
Then can this be `long`?

Reply:
Technically yes it can be long. It is that `SourceLocation` of Google Cloud Logging uses `Long` instead of `long` and I do not understand the intention here. It is updated to `long` now.

Reply:
If Cloud Logging uses `Long` then this should follow suit and use `Long`. There has to be a reason for them to do it and we don't always need to know, what we do need to know is that the output type is `Long` which means it is nullable.

Reply:
OK. It is fixed.